### PR TITLE
Remove PodHostIPs feature gates in 1.32

### PR DESCRIPTION
**Pull request by liangyuanpeng**
#### What type of PR is this?

/kind cleanup
#### What this PR does / why we need it:

the featurgate PodHostIPs  have been GA since v1.30, and can be delete now.

/assign @wzshiming

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
The PodHostIPs  features graduated in v1.30.  The associated feature gates have been removed.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -603,15 +603,6 @@ const (
 	// sandbox creation and network configuration completes successfully
 	PodReadyToStartContainersCondition featuregate.Feature = "PodReadyToStartContainersCondition"
 
-	// owner: @wzshiming
-	// kep: http://kep.k8s.io/2681
-	// alpha: v1.28
-	// beta: v1.29
-	// GA: v1.30
```
Comment (SergeyKanzhelev) [inline pkg/features/kube_features.go:610]: @wzshiming next time please remove code using the feature gate when promoting to GA.

**Comment by SergeyKanzhelev** [comment]
@liangyuanpeng  please fix linter - failures are relevant to the PR

File: pkg/features/kube_features.go (hunk 1 above)
Comment (wzshiming) [reply pkg/features/kube_features.go:610]: Okay, get it, thanks.

File: test/e2e_node/pod_host_ips.go [hunk 2]
```diff
@@ -111,10 +108,6 @@ var _ = common.SIGDescribe("Pod Host IPs", framework.WithSerial(), func() {
 		})
 
 		ginkgo.It("should provide hostIPs as an env var", func(ctx context.Context) {
```
Comment (wzshiming) [inline test/e2e_node/pod_host_ips.go:110]: I think this test can be moved to join the other `Downward API` tests.

https://github.com/kubernetes/kubernetes/blob/f2cfbf44b1fb482671aedbfff820ae2af256a389/test/e2e/common/node/downwardapi.go#L37

File: test/e2e_node/pod_host_ips.go
Comment (wzshiming) [inline test/e2e_node/pod_host_ips.go:1]: https://github.com/kubernetes/kubernetes/blob/f2cfbf44b1fb482671aedbfff820ae2af256a389/test/e2e/common/node/downwardapi.go
https://github.com/kubernetes/kubernetes/blob/f2cfbf44b1fb482671aedbfff820ae2af256a389/test/e2e/network/dual_stack.go

I think this test can be moved to where the other related tests are, that's where this test started, and now that this is GA, I think it's best to bring it back.

Comment (liangyuanpeng) [reply test/e2e_node/pod_host_ips.go:1]: Do you mean the test of `validateDownwardAPIHostIPs`? 

for `pod_host_ips.go` file, this PR just delete `!utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodHostIPs)` and have not tests deleted.

File: test/e2e_node/pod_host_ips.go (hunk 2 above)
Comment (SergeyKanzhelev) [inline test/e2e_node/pod_host_ips.go:110]: which test lane runs this test? I don't see it being marked as NodeConformance, which I would assume the right categorization. But I cannot see it in https://testgrid.k8s.io/sig-node-containerd#node-e2e-unlabelled either.

File: test/e2e_node/pod_host_ips.go
Comment (liangyuanpeng) [reply test/e2e_node/pod_host_ips.go:1]: I got it, i think you're talking about add this tests to downwardapi and dualstack,  So do you think we need to keep this test?

**Comment by googs1025** [comment]
hi, @liangyuanpeng 
We can update this code and now solve this flake test problem.

https://github.com/kubernetes/kubernetes/issues/126801

File: test/e2e_node/pod_host_ips.go (hunk 2 above)
Comment (liangyuanpeng) [reply test/e2e_node/pod_host_ips.go:110]: Seems like it's skiped by `Serial`

https://github.com/kubernetes/test-infra/blob/6b9389ee5a64a70556b31b4bf7d7557a831f1fc9/config/jobs/kubernetes/sig-node/containerd.yaml#L645

Comment (liangyuanpeng) [reply test/e2e_node/pod_host_ips.go:110]:
```yaml
...
      - --test_args=--nodes=1 --skip="\[Flaky\]|\[NodeConformance\]|\[Conformance\]|\[Serial\]|...
...
```

File: test/featuregates_linter/test_data/unversioned_feature_list.yaml [hunk 3]
```diff
@@ -574,12 +574,6 @@
     lockToDefault: true
     preRelease: GA
     version: ""
-- name: PodHostIPs
-  versionedSpecs:
-  - default: true
-    lockToDefault: true
-    preRelease: GA
-    version: ""
```
Comment (pacoxu) [inline test/featuregates_linter/test_data/unversioned_feature_list.yaml:582]: After this is migrated to the versioned feature list in https://github.com/kubernetes/kubernetes/pull/126791/files#diff-530f68e01a51f788bd0ad558bde87a4a378bd3e40e3811d097029f6aa109c7a7R272-R276, I think we need to rebase on that.


```
	PodHostIPs: {
		{Version: version.MustParse("1.28"), Default: false, PreRelease: featuregate.Alpha},
		{Version: version.MustParse("1.29"), Default: true, PreRelease: featuregate.Beta},
		{Version: version.MustParse("1.30"), Default: true, PreRelease: featuregate.GA, LockToDefault: true}, 
		{Version: mustParseVersion("1.32"), Default: false, PreRelease: Removed},

	},
```

File: test/e2e_node/pod_host_ips.go [hunk 4]
```diff
@@ -29,21 +29,18 @@ import (
...
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 	e2epodoutput "k8s.io/kubernetes/test/e2e/framework/pod/output"
-	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 	"k8s.io/kubernetes/test/e2e/network/common"
 	imageutils "k8s.io/kubernetes/test/utils/image"
 	admissionapi "k8s.io/pod-security-admission/api"
 )
 
-var _ = common.SIGDescribe("Pod Host IPs", framework.WithSerial(), func() {
+var _ = common.SIGDescribe("Pod Host IPs", func() {
```
Comment (SergeyKanzhelev) [inline test/e2e_node/pod_host_ips.go:43]: Can you please make it NodeConformance since it doesn't require any special set up

**Comment by SergeyKanzhelev** [comment]
@liangyuanpeng do you still plan to move this one forward?

**Comment by liangyuanpeng** [comment]
yes, I thought we would finish migrating featuregate versiond first, but now featuregate versiond is on hold. Now we can move forward. Thank you for your patience. Let me update it later.

**Comment by haircommander** [comment]
/milestone v1.32

(to track)

**Review by thockin — COMMENTED** [review]
LGTM overall - a few open issues. 

Thanks!

**Comment by liangyuanpeng** [comment]
I am struggling with local testing and have yet to get rid of the error:

```shell
I0914 08:38:21.628603  204905 e2e_node_suite_test.go:249] Wait for the node to be ready
  [FAILED] in [SynchronizedBeforeSuite] - k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:333 @ 09/14/24 08:40:21.629
  << Timeline

  [FAILED] Timed out after 120.000s.
  Expected success, but got an error:
      <*fmt.wrapError | 0xc000af2160>: 
      failed to get node: empty node list: &NodeList{ListMeta:{ 96  <nil>},Items:[]Node{},}
      {
          msg: "failed to get node: empty node list: &NodeList{ListMeta:{ 96  <nil>},Items:[]Node{},}",
          err: <*errors.errorString | 0xc00088a370>{
              s: "empty node list: &NodeList{ListMeta:{ 96  <nil>},Items:[]Node{},}",
          },
      }
  In [SynchronizedBeforeSuite] at: k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:333 @ 09/14/24 08:40:21.629
```

running with `make test-e2e-node REMOTE=false FOCUS="Pod Host IPs" SKIP=""`

BTW, I'm going on vacation next, so I may be offline for a while until 9/23.

File: test/e2e/common/node/downwardapi.go [hunk 5]
```diff
@@ -110,6 +110,32 @@ var _ = SIGDescribe("Downward API", func() {
 		testDownwardAPI(ctx, f, podName, env, expectations)
 	})
 
+	/*
+	   Release: v1.32
+	   Testname: DownwardAPI, environment for hostIPs
+	   Description: Downward API MUST expose Pod and Container fields as environment variables. Specify hostIPs as environment variable in the Pod Spec are visible at runtime in the container.
+	*/
+	framework.ConformanceIt("should provide hostIPs as an env var", f.WithNodeConformance(), func(ctx context.Context) {
```
Comment (SergeyKanzhelev) [inline test/e2e/common/node/downwardapi.go:118]: regenerate conformance yaml file: https://github.com/kubernetes/kubernetes/blob/master/test/conformance/README.md

Comment (SergeyKanzhelev) [inline test/e2e/common/node/downwardapi.go:118]: conformance test in this folder do not need `WithNodeConformance`. NodeConformance makes most sense for the node-related tests, not to API tests

File: test/e2e_node/pod_host_ips.go
Comment (thockin) [reply test/e2e_node/pod_host_ips.go:1]: Shouldn't this file be named `..._test.go` ?

**Comment by SergeyKanzhelev** [comment]
> I am struggling with local testing and have yet to get rid of the error:
> 
> ```shell
> I0914 08:38:21.628603  204905 e2e_node_suite_test.go:249] Wait for the node to be ready
>   [FAILED] in [SynchronizedBeforeSuite] - k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:333 @ 09/14/24 08:40:21.629
>   << Timeline
> 
>   [FAILED] Timed out after 120.000s.
>   Expected success, but got an error:
>       <*fmt.wrapError | 0xc000af2160>: 
>       failed to get node: empty node list: &NodeList{ListMeta:{ 96  <nil>},Items:[]Node{},}
>       {
>           msg: "failed to get node: empty node list: &NodeList{ListMeta:{ 96  <nil>},Items:[]Node{},}",
>           err: <*errors.errorString | 0xc00088a370>{
>               s: "empty node list: &NodeList{ListMeta:{ 96  <nil>},Items:[]Node{},}",
>           },
>       }
>   In [SynchronizedBeforeSuite] at: k8s.io/kubernetes/test/e2e_node/e2e_node_suite_test.go:333 @ 09/14/24 08:40:21.629
> ```
> 
> running with `make test-e2e-node REMOTE=false FOCUS="Pod Host IPs" SKIP=""`
> 
> BTW, I'm going on vacation next, so I may be offline for a while until 9/23, and then I'll get on with it.

check the kubelet.log file for the specific error.


Also, it seems like all tests are passing now in CI except the verification, that is due to conformance.yaml wasn't regenerated

**Comment by kannon92** [comment]
Please rebase this PR. It would be great to get this in for 1.32.

**Comment by wendy-ha18** [comment]
Hello @liangyuanpeng @SergeyKanzhelev @kannon92

Friendly reminder that code freeze is starting 02:00 UTC Friday November 8th 2024 (about 3 weeks from now), and while there is still time, we want to ensure that each PR has a chance to be merged on time.

Given this timeline and capacity, will this continue to be aimed for the 1.32 release?
Thanks! 😄

**Comment by thockin** [comment]
ping @liangyuanpeng - one more try before code-freeze

**Comment by thockin** [comment]
Do you want someone to take this over?

**Comment by wendy-ha18** [comment]
Hey @liangyuanpeng @thockin @kannon92 and other contributors working on this PR
Thanks for your effort so far, just a last reminder from my side that the code freeze is starting 02:00 UTC Friday November 8th 2024 (less than 1 week from now).

Please make sure the PR for has both lgtm and approved labels before the code freeze. Thanks!

**Comment by thockin** [comment]
https://github.com/kubernetes/kubernetes/pull/128634 takes this over

**Pull request closed without merging**